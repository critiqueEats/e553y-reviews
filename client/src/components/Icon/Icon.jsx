import React from 'react';
import PropTypes from 'prop-types'
import './reviews-icons.svg';

const Icon = ({name, width, height, fill}) => (
    <svg className={`icon-${name}`} width={width} height={height} fill={fill}>
      <use xlinkHref={`#reviews-icons_${name}`} />
    </svg>
  )
Icon.propType = {
  name: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
}

Icon.defaultProps = {
  height: 18,
  width: 18,
  fill: '#000'
}
export default Icon