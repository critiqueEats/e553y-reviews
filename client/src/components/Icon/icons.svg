<svg version="1.1" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <symbol id="friends" height="18" viewBox="0 0 18 18" width="18">
      <g>
        <path d="M7.904 9.43l-2.098 4.697a.9.9 0 0 1-1.612 0L2.096 9.43a.902.902 0 0 1 .806-1.305h4.196c.67 0 1.105.705.806 1.305zM5 7.375a2 2 0 1 1 0-4 2 2 0 0 1 0 4z"></path>
        <path d="M15.904 9.43l-2.098 4.697a.89.89 0 0 1-.806.498.89.89 0 0 1-.806-.498L10.096 9.43a.902.902 0 0 1 .806-1.305h4.195c.67 0 1.106.705.807 1.305zM13 7.375a2 2 0 1 1 0-4 2 2 0 0 1 0 4z" opacity=".502"></path>
      </g>
    </symbol>
    <symbol id="big_review" height="24" viewBox="2 3 20 18" width="24">
      <path d="M21 6a3 3 0 0 0-3-3H6a3 3 0 0 0-3 3v12a3 3 0 0 0 3 3h12a3 3 0 0 0 3-3V6zm-5.88 10.428l-3.16-1.938-3.05 2.01.59-3.457L7 10.596l3.457-.505L11.96 6.5l1.582 3.59 3.458.506-2.5 2.447.62 3.385z">
      </path>
    </symbol>
    <symbol id="review" height="18" viewBox="0 0 18 18" width="18">
      <path d="M13 3H5c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-1.505 9.643l-2.526-1.55L6.526 12.7 7 9.934 5 7.977l2.766-.404L8.97 4.7l1.264 2.873L13 7.977l-2 1.957.495 2.71z"></path>
    </symbol>
    <symbol id="camera" height="18" viewBox="0 0 18 18" width="18">
      <path d="M15 15H3a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2h2a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2zM9 5a4 4 0 1 0 0 8 4 4 0 0 0 0-8zm0 6.5a2.5 2.5 0 1 1 0-5 2.5 2.5 0 0 1 0 5z"></path>
    </symbol>
    <symbol id="share" height="18" viewBox="0 0 18 18" width="18">
      <path d="M17.714 6.43L13 10.356v-3.03c-1 0-5.097 1.47-6.286 3.62.274-3.08 4.286-5.5 6.286-5.5V2.5l4.714 3.93zM3 4v10h11v-2.5l1-1V15H2V3h8.5l-1 1H3z"></path>
    </symbol>
    <symbol id="embed" height="18" viewBox="0 0 18 18" width="18">
      <path d="M12.503 13.76a.717.717 0 0 1-.526.23.717.717 0 0 1-.526-.23.822.822 0 0 1 0-1.117l3.444-3.652-3.444-3.65a.822.822 0 0 1 0-1.116.714.714 0 0 1 1.053 0l4.495 4.77-4.495 4.767zm-5.954 0a.717.717 0 0 1-.528.23.718.718 0 0 1-.526-.23L1.002 8.99l4.495-4.768a.714.714 0 0 1 1.052 0c.29.308.29.808 0 1.116L3.105 8.99l3.442 3.653c.29.308.29.808 0 1.116z"></path>
    </symbol>
    <symbol id="compliment" height="18" viewBox="0 0 18 18" width="18">
      <path d="M14.5 6.75a5.5 5.5 0 0 0-11 0 5.495 5.495 0 0 0 2.993 4.892L5.5 16.75l3.505-2 3.495 2-.993-5.108A5.495 5.495 0 0 0 14.5 6.75zm-5.5 4c-2.206 0-4-1.794-4-4s1.794-4 4-4 4 1.794 4 4-1.794 4-4 4z"></path>
    </symbol>
    <symbol id="speech" height="18" viewBox="0 0 18 18" width="18">
      <path d="M2 4v6a2 2 0 0 0 2 2h1v3l4-3h5a2 2 0 0 0 2-2V4a2 2 0 0 0-2-2H4a2 2 0 0 0-2 2z"></path>
    </symbol>
    <symbol id="following" height="18" viewBox="0 0 18 18" width="18">
      <path d="M16 13l-4 3v-2H9l1-2h2v-2l4 3zm-8.925 2.383c-.443.817-1.707.817-2.15 0L2.128 9.605C1.728 8.868 2.31 8 3.203 8h5.594c.893 0 1.474.868 1.075 1.605l-2.797 5.778zM6 7a2.5 2.5 0 1 1 0-5 2.5 2.5 0 0 1 0 5z"></path>
    </symbol>
    <symbol id="useful_outline" height="18" viewBox="0 0 18 18" width="18">
      <path d="M9 17c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM9 2C5.14 2 2 5.14 2 9s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm2 8.392V12H7v-1.608a3.982 3.982 0 0 1-2-3.445 4 4 0 0 1 8 0c0 1.477-.81 2.752-2 3.445zM8 5.25a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5zm1.003 9.747h-.006A1.997 1.997 0 0 1 7 13h4a1.997 1.997 0 0 1-1.997 1.997z"></path>
    </symbol>
    <symbol id="funny_outline" height="18" viewBox="0 0 18 18" width="18">
      <path d="M9 17c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM9 2C5.14 2 2 5.14 2 9s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm0 12a4.87 4.87 0 0 1-4.787-4h9.574A4.87 4.87 0 0 1 9 14zm2.5-5.625a1.376 1.376 0 1 1 0-2.75 1.376 1.376 0 0 1 0 2.75zm-5 0a1.376 1.376 0 1 1 0-2.75 1.376 1.376 0 0 1 0 2.75z"></path>
    </symbol>
    <symbol id="cool_outline" height="18" viewBox="0 0 18 18" width="18">
      <path d="M9 17c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM9 2C5.14 2 2 5.14 2 9s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm6.026 5.335C14.766 8.797 13.5 10 11.986 10h-.003c-1.218 0-2.282-.764-2.767-1.813-.088-.19-.344-.242-.432-.052C8.3 9.185 7.234 10 6.016 10h-.003C4.5 10 3.195 8.83 2.973 7.35l-.093-.84c-.053-.242.192-.51.477-.51h11.286c.294 0 .508.332.477.56l-.094.775zm-2.068 4.154A4.28 4.28 0 0 1 9 14.144a4.28 4.28 0 0 1-3.958-2.657A6.81 6.81 0 0 0 9 12.753a6.81 6.81 0 0 0 3.958-1.265z"></path>
    </symbol>
  </defs>
</svg>
