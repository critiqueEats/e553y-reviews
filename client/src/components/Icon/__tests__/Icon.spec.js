import React from 'react';
import {shallow, mount, configure, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
jest.mock('../reviews-icons.svg', ()=> 'svgfile'); 

import Icon from '../Icon.jsx';
describe('Icon component', function() {
  let testProps = {
    name: 'name',
    width: 18,
    height: 18,
    fill: 'red'
  }
  it('should generate matching html snap-shot', function() {
    let wrapper = shallow(<Icon  name="name"/>)
    expect(wrapper.html()).toMatchSnapshot()
  
  })
  it('should render an svg element', function() {
    
    let wrapper = shallow(<Icon {...testProps} />)

    expect(wrapper.name()).toBe('svg')
  })
  
  it('should render use element', function() {
    let wrapper = shallow(<Icon {...testProps} />)

    expect(wrapper.find('use')).toHaveLength(1)
  })
})