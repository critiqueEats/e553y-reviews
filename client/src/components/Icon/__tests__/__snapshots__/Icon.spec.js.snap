// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should generate matching html snap-shot 1`] = `"<svg class=\\"icon-name\\" width=\\"18\\" height=\\"18\\" fill=\\"#000\\"><use xlink:href=\\"#reviews-icons_name\\"></use></svg>"`;
